### DISCUSSION TITLE
Is it possible to send a new private tab to an existing private window from CLI?

### DISCUSSION THREAD

**Discussion by carlosgarciafm** [qutebrowser/qutebrowser#6320] [Q&A] [answered]
The behaviour I'm looking for is for qutebrowser to attach the new private tab to an existing private instance if one exists, otherwise, create a new private window with the given URL.

I've been skimming the docs and issues and found that you can "give" tabs from within qutebrowser using the command `tab-give -p {URL}`. Invoking qutebrowser from CLI like `qutebrowser --target private-window {URL}` always creates a new private window with the given URL. You can also attach new tabs from CLI by simply running `qutebrowser {URL}` with no extra arguments which defaults to `--target tab`. Using the latter, however, has the limitation of only be able to target windows that match 1 of 4 criteria: _first-opened_, _last-opened_, _last-focused_ (default) and _last-visible_.

The closest I'm able to get is by making sure a private window exists and it was the last one with focus and then running `qutebrowser {URL}`; this only works because qutebrowser defaults to sending a tab to the last instance with focus.

**Comment by The-Compiler** [comment, marked as answer]
Sorry for the radio silence, this got lost in my inbox for a while. I don't think that's possible at the moment. I opened #6369 with some ideas.

**Reply by carlosgarciafm** [reply]
Don't mind. I'll check the issue, thanks!
